### PR TITLE
  refactor(valkey): use async DocumentStore mixin tests

### LINKED ISSUE #3053: Use async DocumentStore mixin tests in Valkey
As part of [deepset-ai/haystack#10799](https://github.com/deepset-ai/haystack/issues/10799), async test mixin classes are being added to `haystack.testing.document_store`. Once those are in place, the Valkey integration should inherit the relevant mixins to avoid maintaining duplicate test code.

## What needs to be done

In `integrations/valkey/tests/test_document_store_async.py`, update the async test class to inherit the appropriate mixin classes:

```python
from haystack.testing.document_store import (
    CountDocumentsAsyncTest,
    WriteDocumentsAsyncTest,
    DeleteDocumentsAsyncTest,
    DeleteAllAsyncTest,
    DeleteByFilterAsyncTest,
    UpdateByFilterAsyncTest,
    CountDocumentsByFilterAsyncTest,
    CountUniqueMetadataByFilterAsyncTest,
    GetMetadataFieldsInfoAsyncTest,
    GetMetadataFieldMinMaxAsyncTest,
    GetMetadataFieldUniqueValuesAsyncTest,
)
```

> **Note:** The existing Valkey async tests use non-standard naming (e.g. `test_async_write_and_count_documents` instead of `test_write_documents_async`). When inheriting the mixins, ensure the mixin tests don't overlap in intent with the remaining store-specific tests.

## Tests that can be removed (covered by mixins)

- `test_async_write_and_count_documents` → covered by `CountDocumentsAsyncTest` + `WriteDocumentsAsyncTest`
- `test_async_write_and_delete_documents` → covered by `DeleteDocumentsAsyncTest`
- `test_async_delete_all_documents` → covered by `DeleteAllAsyncTest`
- `test_async_delete_all_documents_empty_store` → covered by `DeleteAllAsyncTest`
- `test_delete_by_filter_async`
- `test_update_by_filter_async`
- `test_count_documents_by_filter_async`
- `test_count_unique_metadata_by_filter_async`
- `test_get_metadata_fields_info_async`
- `test_get_metadata_field_min_max_async`
- `test_get_metadata_field_unique_values_async`

## Tests to keep (Valkey-specific)

- `test_async_write_exceed_batch_size`
- `test_async_write_exact_batch_size`
- `test_async_write_multiple_full_batches`
- `test_async_overwrite_documents`
- `test_async_write_large_batch_performance`
- `test_async_search_by_embedding_no_limit`
- `test_async_search_by_embedding_with_limit`
- `test_async_search_by_embedding_with_category_filter`
- `test_async_search_by_embedding_with_numeric_filter`
- `test_async_search_by_embedding_with_or_filter`
- `test_async_search_by_embedding_with_in_filter`
- `test_async_similarity_scores_are_set_correctly`
- `test_async_filter_by_meta_score`
- `test_async_search_with_meta_score_filter`
- `test_get_metadata_field_min_max_empty_store_async` (Valkey-specific empty-store edge case)
- `test_get_metadata_field_unique_values_with_search_term_async` (Valkey-specific search term behaviour)

## Acceptance criteria

- [ ] The async test class inherits the relevant async mixin classes
- [ ] The `document_store` async fixture is correctly wired up
- [ ] All duplicate tests listed above are removed
- [ ] All Valkey-specific tests are preserved
- [ ] The full test suite passes

### PR DESCRIPTION
**Pull request by SyedShahmeerAli12**
Closes #3053

  Refactors `integrations/valkey/tests/test_document_store_async.py` to inherit from the async  
  mixin classes introduced in deepset-ai/haystack#10975.

  - Removes 10 duplicate tests now covered by the mixins
  - Keeps Valkey-specific tests (batch operations, embedding search, scoring, metadata)
  - Overrides `WriteDocumentsAsyncTest` methods for unsupported duplicate policies (`FAIL`,     
  `SKIP`) since ValkeyDocumentStore only supports `NONE` and `OVERWRITE`

### CODE REVIEW

File: integrations/valkey/tests/test_document_store_async.py [hunk 1]
```diff
@@ -10,19 +10,56 @@
 import pytest_asyncio
 from haystack.dataclasses import ByteStream, Document
 
+pytest.importorskip("haystack.testing.document_store_async", reason="Requires haystack with async mixin test classes")
```
Comment (davidsbatista) [inline integrations/valkey/tests/test_document_store_async.py]: We can remove this since `haystack-ai` is already a dependency on the `pyproject.toml`

File: integrations/valkey/pyproject.toml [hunk 2]
```diff
@@ -59,6 +59,7 @@ dependencies = [
     "pytest-asyncio",
     "pytest-cov",
     "pytest-rerunfailures",
+    "haystack-ai @ git+https://github.com/deepset-ai/haystack.git@main",
```
Comment (davidsbatista) [inline integrations/valkey/pyproject.toml]: we can also remove this one

**Review by davidsbatista — COMMENTED** [review]
looks good - left some minor comments

**Comment by davidsbatista** [comment]
currently on hold until `2.28.0`

**Review by davidsbatista — APPROVED** [review]
Looks good!
